src/Operations.js

```diff
@@ -726,9 +726,15 @@ function maxCompare(comparator, a, b) {
 }
 
 
-export function zipWithFactory(keyIter, zipper, iters) {
+export function zipWithFactory(keyIter, zipper, iters, zipAll) {
   var zipSequence = makeSequence(keyIter);
-  zipSequence.size = new ArraySeq(iters).map(i => i.size).min();
+  var iterSizes = new ArraySeq(iters).map(i => i.size);
+  var zipSeqSize = !!zipAll ? iterSizes.max() : iterSizes.min();
+  if (zipAll) {
```

Comment:
Yeah this concat could be expensive. I think the logic to decide when to end the iteration should exist within the iterator.